src/ansible_navigator/configuration_subsystem/navigator_post_processor.py

```diff
@@ -66,31 +72,78 @@ class VolumeMountOption(Enum):
     z = "z"  # pylint: disable=invalid-name
 
 
+V = TypeVar("V", bound="VolumeMount")  # pylint: disable=invalid-name
+
+
+class VolumeMountError(Exception):
+    """Custom exception raised when building VolumeMounts."""
+
+
 @dataclass
 class VolumeMount:
     """Describes EE volume mounts."""
 
-    #: The name of the config option requiring this volume mount.
-    calling_option: str
-
-    #: The source path of the volume mount.
-    src: str
-
-    #: The destination path in the container for the volume mount.
-    dest: str
+    fs_source: str
+    """The source file system path of the volume mount"""
+    fs_destination: str
+    """The destination file system path in the container for the volume mount"""
+    settings_entry: str
+    """The name of the settings entry requiring this volume mount"""
+    source: C
+    """The settings source for this volume mount"""
+    options_string: str = ""
+    """Comma delimited options"""
+    _options: List[VolumeMountOption] = field(default_factory=list)
+    """Options for the bind mount"""
+
+    def __post_init__(self):
+        """Post process the ``VolumeMount`` and perform sanity checks.
+
+        :raises VolumeMountError: When a viable VolumeMount cannot be created
+        """
+        # pylint: disable=too-many-branches
+        errors = []
+        # Validate the source
+        if isinstance(self.fs_source, str):
+            if self.fs_source == "":
+                errors.append("Source not provided.")
+            elif not Path(self.fs_source).exists():
+                errors.append(f"Source: '{self.fs_source}' does not exist.")
+        else:
+            errors.append(f"Source: '{self.fs_source}' is not a string.")
 
-    #: Options for the bind mount.
-    options: List[VolumeMountOption] = field(default_factory=list)
+        # Validate the destination
+        if isinstance(self.fs_destination, str):
+            if self.fs_destination == "":
+                errors.append("Destination not provided.")
+        else:
+            errors.append(f"Destination: '{self.fs_destination} is not a string.")
+
+        # Validate and populate _options
+        if isinstance(self.options_string, str):
+            if not self.options_string == "":
+                options = []
+                option_values = [o.value for o in VolumeMountOption]
+                for option in self.options_string.split(","):
+                    if option not in option_values:
+                        errors.append(
+                            f"Unrecognized label: '{option}',"
+                            " available labels include {oxfordcomma(option_values)}.",
```

Comment:
I still don't get why we're calling these labels. They're options:

- [`podman run` manpage](https://docs.podman.io/en/latest/markdown/podman-run.1.html#volume-v-source-volume-host-dir-container-dir-options)
- [`docker run` docs](https://docs.docker.com/engine/reference/run/#volume-shared-filesystems)

Some of the options happen to control relabeling behavior. But they aren't labels.

```suggestion
                            f"Unrecognized option: '{option}',"
                            " available options include {oxfordcomma(option_values)}.",
```